### PR TITLE
Extend CEL validation

### PR DESCRIPTION
**Pull request by werdes72**
**Description**

Changes proposed in this pull request:

- Use CEL validation for FQDN hosts
- Fix Docker warning
- Use CEL validation for gateway name
- Check if gateway defined in the APIRule exists
- Tests

**Pre-Merge Checklist**

- [ ] As a PR reviewer, verify code coverage and evaluate if it is acceptable.

**Related issues**
#1149

### CODE REVIEW

File: controllers/gateway/api_controller_integration_test.go [hunk 1]
```diff
@@ -1395,6 +1395,330 @@ var _ = Describe("APIRule Controller", Serial, func() {
...
+			// when
+			Expect(c.Create(context.Background(), svc)).Should(Succeed())
+			Expect(c.Create(context.Background(), apiRule)).Should(Succeed())
+			defer func() {
+				apiRulev2alpha1Teardown(apiRule)
+				serviceTeardown(svc)
+			}()
+		})
+
+		It("should not create an APIRule with host name without domain", func() {
```
Comment (barchw) [inline controllers/gateway/api_controller_integration_test.go:1509]: There's a lot of stuff duplicated in those tests, maybe we could make them into a table, or at least create some setup func that would take just the host?

**Review by barchw — CHANGES_REQUESTED** [review]

**Review by barchw — CHANGES_REQUESTED** [review]
Needs docs, otherwise LGTM

**Review by barchw — DISMISSED** [review]

File: docs/user/custom-resources/apirule/v2alpha1/04-10-apirule-custom-resource.md [hunk 2]
```diff
@@ -13,38 +13,38 @@ This table lists all parameters of APIRule `v2alpha1` CRD together with their de
...
+|:-------------------------------------------------|:--------:|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------|
+| **gateway**                                      | **YES**  | Specifies the Istio Gateway it must reference actual gateway on the cluster.                                                                                                                                                                                                          | Must be in `namespace/gateway` format, namespace and gateway cannot be longer than 63 characters each. |
```
Comment (nataliasitko) [inline docs/user/custom-resources/apirule/v2alpha1/04-10-apirule-custom-resource.md:18]:
```suggestion
| **gateway**                                      | **YES**  | Specifies the Istio Gateway. The value must reference an actual Gateway in the cluster.                                                                                                                                                                                                          | It must be in the `namespace/gateway` format. The namespace and the Gateway cannot be longer than 63 characters each. |
```

File: docs/release-notes/2.5.0.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+## New Features
+
+- Add gateway validation [#1210](https://github.com/kyma-project/api-gateway/pull/1210)
```
Comment (nataliasitko) [inline docs/release-notes/2.5.0.md:3]:
```suggestion
- Add Gateway validation [#1210](https://github.com/kyma-project/api-gateway/pull/1210)
```

**Review by nataliasitko — APPROVED** [review]

**Pull request merged**